lib/set.js

```diff
@@ -59,13 +67,12 @@ Set.prototype.addMigration = function(title, up, down){
  */
 
 Set.prototype.save = function(fn){
-  var self = this
-    , json = JSON.stringify(this);
-  fs.writeFile(this.path, json, function(err){
-    if (err) return fn(err);
-
+  var self = this;
+  this.state.save({
```

Comment:
A possibility could be to extend this pluggable state class such that it stored and loaded the entire state passed to it - instead of complying with the old implementation - and also make other parts of the migrate package pluggable. That way someone could make conflict-resolution-strategy plugins for specific databases.

Do you know any other migration frameworks that handle cases with "conflicting" migrations automatically?